### PR TITLE
Add import and download log info

### PR DESCRIPTION
**Pull request by bsteuber-fdc**

### CODE REVIEW

File: core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java [hunk 1]
```diff
@@ -56,16 +56,21 @@ default String getControlFileEnding() {
...
     public static final String DEFAULT_CONFIG_DOWNLOAD_DIR_PATH = "/tmp/downloading";
     public static final String DEFAULT_CONFIG_CONTROL_FILE_ENDING = ".ctl";
 
-
     public static final class ConfigurationImpl implements Configuration {
 
         private String downloadDirPath = DEFAULT_CONFIG_DOWNLOAD_DIR_PATH;
         private String controlFileEnding = DEFAULT_CONFIG_CONTROL_FILE_ENDING;
+        private String logFileEnding = DEFAULT_CONFIG_LOG_FILE_ENDING;
+        public static final String DEFAULT_CONFIG_CONTROL_FILE_ENDING = ".ctl";
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java]: why did you change this?

File: core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java [hunk 2]
```diff
@@ -56,16 +56,21 @@ default String getControlFileEnding() {
             return ".ctl";
         }
 
+        default String getLogFileEnding() {
+            return ".log";
+        }
     }
 
     public static final String DEFAULT_CONFIG_DOWNLOAD_DIR_PATH = "/tmp/downloading";
     public static final String DEFAULT_CONFIG_CONTROL_FILE_ENDING = ".ctl";
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java:65]: Why did'nt you introduce your static final constant next to this one?

File: core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java (hunk 1 above)
Comment (bsteuber-fdc) [inline core/src/main/java/com/freiheit/fuava/simplebatch/fsjobs/downloader/CtlDownloaderJob.java]: oups

File: core/src/main/java/com/freiheit/fuava/simplebatch/logging/ImportFileJsonLoggingListener.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public void onProcessingResult( Result<FetchedItem<ControlFile>, ResultStatistics> result ) {
+        final String logFileName = result.getInput().getValue().getLogFileName();
+        final String dir = result.isSuccess()
+            ? archivedDir
+            : failedDir;
+        BatchJsonLogger l = new BatchJsonLogger( Paths.get( dir, logFileName ) );
+        l.logImportEnd( result.isSuccess() );
+        ;
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/logging/ImportFileJsonLoggingListener.java]: superfluous

File: core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingFailureProcessor.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        this.controlFileEnding = controlFileEnding;
+        this.logFileEnding = logFileEnding;
+    }
+
+    @Override
+    public Iterable<Result<FetchedItem<Input>, Output>> process( Iterable<Result<FetchedItem<Input>, Output>> iterable ) {
+        List<String> failedInputs = new ArrayList<String>();
+        for ( Result<FetchedItem<Input>, Output> res : iterable ) {
+            if ( res.isFailed() ) {
+                failedInputs.add( res.getInput().getValue().toString() );
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingFailureProcessor.java]: I am not sure that you can always rely on input value being non-null

File: core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingProcessor.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            Path logFile;
+            String failedPrefix = "";
+            if ( res.isFailed() ) {
+                failedPrefix = BatchJsonLogger.nextFailedDownloadsName();
+                logFile = Paths.get( dirName, failedPrefix + logFileEnding );
+            } else {
+                logFile = Paths.get( res.getOutput().getDataFile() + logFileEnding );
+            }
+            BatchJsonLogger l = new BatchJsonLogger( logFile );
+            l.logWriteEnd( res.getInput().getValue().toString(), res.isSuccess() );
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingProcessor.java]: input.value could be null

File: core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingProcessor.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package com.freiheit.fuava.simplebatch.processor;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.freiheit.fuava.simplebatch.fetch.FetchedItem;
+import com.freiheit.fuava.simplebatch.logging.BatchJsonLogger;
+import com.freiheit.fuava.simplebatch.result.Result;
+
+public class BatchJsonLoggingProcessor<Input>
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/processor/BatchJsonLoggingProcessor.java]: What is the distinction between BatchJsonLogginProcessor and BatchJsonLoggingFailureProcessor?

File: core/src/main/java/com/freiheit/fuava/simplebatch/processor/Processors.java [hunk 7]
```diff
@@ -154,17 +150,25 @@
...
             final String dirName,
             final String controlFileEnding,
-            final FileOutputStreamAdapter<List<Input>, List<Output>> adapter
-            ) {
-        return new BatchProcessor<Input, Output, ControlFilePersistenceOutputInfo>( controlledFileWriter( dirName,
-                controlFileEnding, adapter ) );
+            final String logFileEnding,
+            final FileOutputStreamAdapter<List<FetchedItem<Input>>, List<Output>> adapter ) {
+        return Processors.compose(
+                new BatchProcessor<FetchedItem<Input>, Output, ControlFilePersistenceOutputInfo>(
```
Comment (kkalass) [inline core/src/main/java/com/freiheit/fuava/simplebatch/processor/Processors.java:161]: please add a comment here that informs about the fact, that the BatchProcessor here only processes successful items. Or maybe you can rename it to something like SuccessItemsBatchProcessor ?

File: core/src/test/java/com/freiheit/fuava/simplebatch/BatchTestDirectory.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    public String getArchiveDir() {
+        return testDirArchive;
+    }
+
+    public String getFailsDir() {
+        return testDirFails;
+    }
+
+    public BatchTestDirectory( String dirName ) {
+        testDirBase = "/tmp/" + dirName + "/" + System.currentTimeMillis();
```
Comment (kkalass) [inline core/src/test/java/com/freiheit/fuava/simplebatch/BatchTestDirectory.java]: please use Files.createTempDir() instead
